File: Fines2Model3/Values/Disposicion.cs
﻿namespace SqlOrganize.Sql.Fines2Model3
{
    public class DisposicionValues : EntityVal
    {
        public DisposicionValues(Db _db, string _entity_name, string? _field_id) : base(_db, _entity_name, _field_id)
        {
        }

        public override string ToString()
        {
            var s = "";

            EntityVal? v = GetValuesCache("asignatura");
            if(v != null)
            {
                s += v.GetOrNull("nombre")?.ToString() ?? "?";
                s += " ";
                s += v.GetOrNull("codigo")?.ToString() ?? "?";
                s += " ";
            }

            v = GetValuesCache("planificacion");
            if (v != null)
            {
                s += v.GetOrNull("anio")?.ToString() ?? "?";
                s += "/";
                s += v.GetOrNull("semestre")?.ToString() ?? "?";
            }

            return s.Trim();
        }
    }
}

    

File: Model/Disposicion_.php
<?php

namespace Fines2;

use \Fines2\Disposicion;
use SqlOrganize\Sql\DbMy;
use SqlOrganize\Utils\ValueTypesUtils;


class Disposicion_ extends Disposicion
{

    public function getLabel(): string {
        return ($this->asignatura_?->nombre ?? "?") . " " . 
            ($this->asignatura_?->codigo ?? "?") . " " . 
            ($this->planificacion_?->anio ?? "?"). "/" .
            ($this->planificacion_?->semestre ?? "?") . " " .
            ($this->planificacion_?->plan_?->resolucion ?? "?") . " " . 
            ValueTypesUtils::acronym($this->planificacion_?->plan_?->orientacion ?? "") . " (" .
            ($this->horas_catedra ?? "?") . ")"; 
    }


    

}

